compiler/vm/vm.nim

```diff
@@ -71,34 +74,105 @@ when hasFFI:
 const
   errIllegalConvFromXtoY = "illegal conversion from '$1' to '$2'"
 
+type AddressCheckFailReason = enum
+  acfrOutOfBounds ## Address points to outside the vm's memory
+  acfrTypeMismatch ## It's not legal to write to the target location with handle of given type
+
+func checkValid(mm: VmMemoryManager, a: MemoryHandle, typ: PVmType, r: var AddressCheckFailReason): bool =
```

Comment:
Add `acfrNoError` then return `AddressCheckFailReason` directly instead, given that you have no use for the `r` other than being a return value.